app/models/match_routes/base.rb

```diff
@@ -80,7 +80,15 @@ def first_client_step
     end
 
     def on_or_after_first_client_step?(current_decision)
-      self.class.match_steps[current_decision.class.name] >= first_client_step_number
+      on_first_client_step?(current_decision) || after_first_client_step?(current_decision)
+    end
```

Comment:
Maybe move this to `ClientOpportunityMatch` to minimize the route API? I don't think it really matters tho...